src/brs/BlockchainProcessorImpl.java

```diff
@@ -149,12 +149,19 @@ public BlockchainProcessorImpl(ThreadPool threadPool, BlockService blockService,
     }, Event.BLOCK_PUSHED);
 
     blockListeners.addListener(block -> transactionProcessor.revalidateUnconfirmedTransactions(), Event.BLOCK_PUSHED);
-
+    if (trimDerivedTables) {    
+      blockListeners.addListener(block -> { 
+        if (block.getHeight() % Constants.MAX_ROLLBACK == 0) {
+          if (lastTrimHeight.get() > 0) {   
+            this.derivedTableManager.getDerivedTables().forEach(table -> table.trim(lastTrimHeight.get())); 
+          } 
+        }   
+      }, Event.AFTER_BLOCK_APPLY);  
+    }
     addGenesisBlock();
     if(Boolean.FALSE.equals(propertyService.getBoolean(Props.DB_SKIP_CHECK))) {
       if(checkDatabaseState() != 0) {
-        logger.error("Database is inconsistent, try to pop off 1000 blocks or more or sync from empty. Alternatively add '{} = true' at your own risk.", Props.DB_SKIP_CHECK.getName());
-        System.exit(-1);
+        logger.warn("Database is inconsistent, try to pop off to block height {} or sync from empty.", getMinRollbackHeight());
```

Comment:
same here, but ofc it's a matter of preference

Reply:
Agreed, will be adjusted and merged.